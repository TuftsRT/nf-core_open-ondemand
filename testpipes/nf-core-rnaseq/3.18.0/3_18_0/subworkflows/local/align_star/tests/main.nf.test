nextflow_workflow {

    name "Test Subworkflow ALIGN_STAR"
    script "../main.nf"
    workflow "ALIGN_STAR"
    config "./nextflow.config"

    test("star - no igenomes") {

        setup {
            run("STAR_GENOMEGENERATE") {
                script "../../../../modules/nf-core/star/genomegenerate/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id:'test_fasta' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                    ])
                    input[1] = Channel.of([
                        [ id:'test_gtf' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                    ])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = false

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    file(workflow.out.log_final[0][1]).name,
                    file(workflow.out.log_out[0][1]).name,
                    workflow.out.bai,
                    workflow.out.bam,
                    workflow.out.bam_sorted,
                    workflow.out.bam_transcript,
                    workflow.out.csi,
                    workflow.out.log_progress,
                    workflow.out.fastq,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.orig_bam,
                    workflow.out.stats,
                    workflow.out.tab,
                    workflow.out.versions).match()}
            )
        }
    }

    test("star - with igenomes") {

        setup {
            run("STAR_GENOMEGENERATE_IGENOMES") {
                script "../../../../modules/local/star_genomegenerate_igenomes/main.nf"
                process {
                    """
                    input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                    input[1] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = true

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE_IGENOMES.out.index.map{ [ [id:'star'], it ] }
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    file(workflow.out.log_final[0][1]).name,
                    file(workflow.out.log_out[0][1]).name,
                    workflow.out.bai,
                    workflow.out.bam,
                    workflow.out.bam_sorted,
                    workflow.out.bam_transcript,
                    workflow.out.csi,
                    workflow.out.log_progress,
                    workflow.out.fastq,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.orig_bam,
                    workflow.out.stats,
                    workflow.out.tab,
                    workflow.out.versions).match()}
            )
        }
    }

    test("star - no igenomes - stub") {

        options "-stub"

        setup {
            run("STAR_GENOMEGENERATE") {
                script "../../../../modules/nf-core/star/genomegenerate/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id:'test_fasta' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                    ])
                    input[1] = Channel.of([
                        [ id:'test_gtf' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                    ])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = false

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.bai,
                    workflow.out.bam,
                    workflow.out.bam_sorted,
                    workflow.out.bam_transcript,
                    workflow.out.csi,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.log_final,
                    workflow.out.log_out,
                    workflow.out.log_progress,
                    workflow.out.orig_bam,
                    workflow.out.stats,
                    workflow.out.tab,
                    workflow.out.versions).match()}
            )
        }
    }

    test("star - with igenomes - stub") {

        options "-stub"

        setup {
            run("STAR_GENOMEGENERATE_IGENOMES") {
                script "../../../../modules/local/star_genomegenerate_igenomes/main.nf"
                process {
                    """
                    input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                    input[1] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = true

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE_IGENOMES.out.index.map{ [ [id:'star'], it ] }
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.bai,
                    workflow.out.bam,
                    workflow.out.bam_sorted,
                    workflow.out.bam_transcript,
                    workflow.out.csi,
                    workflow.out.flagstat,
                    workflow.out.idxstats,
                    workflow.out.log_final,
                    workflow.out.log_out,
                    workflow.out.log_progress,
                    workflow.out.orig_bam,
                    workflow.out.stats,
                    workflow.out.tab,
                    workflow.out.versions).match()}
            )
        }
    }
}
