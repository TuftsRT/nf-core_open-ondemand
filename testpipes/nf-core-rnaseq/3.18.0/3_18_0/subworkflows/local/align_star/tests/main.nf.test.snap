{
    "star - with igenomes": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,4cf31974d82bed533f4b6528dff95b50"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,7137528dd300c4ae0f7e3370a5555c79"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b6cec9bba6b04b9b92eddbef128bdfbb"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b6cec9bba6b04b9b92eddbef128bdfbb"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,bad7f610455554f5523eb1ae58ab7afe"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                "versions.yml:md5,5ad5e88ee4e52ccabeef61dcd368d30b",
                "versions.yml:md5,92f25606f0ee994dac8c00fe49676e5a",
                "versions.yml:md5,9611f602102b5aed0fe2b947f9d2270c",
                "versions.yml:md5,b92d6ad60847882b0089f551c5cfc366",
                "versions.yml:md5,d5e8b4d761dd85a303814239077bd666",
                "versions.yml:md5,e3b043ba840d5bddde92f0bf1c91a5fc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-21T14:53:32.322441137"
    },
    "star - no igenomes - stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.toTranscriptome.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.final.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "testXd.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                "versions.yml:md5,57e0ecbc0fd3d1f5cfc9d9ed285826e8",
                "versions.yml:md5,5ad5e88ee4e52ccabeef61dcd368d30b",
                "versions.yml:md5,92f25606f0ee994dac8c00fe49676e5a",
                "versions.yml:md5,9611f602102b5aed0fe2b947f9d2270c",
                "versions.yml:md5,b92d6ad60847882b0089f551c5cfc366",
                "versions.yml:md5,d5e8b4d761dd85a303814239077bd666"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-19T09:49:19.874380799"
    },
    "star - no igenomes": {
        "content": [
            "test.Log.final.out",
            "test.Log.out",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,59e8552f0e50c540d41198496bf34251"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,0d47a5f971c1548eadc1f3c39dab0dbf"
                ]
            ],
            [
                
            ],
            [
                
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,4e917b4befbbe216fd666d06af0298af"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,bad7f610455554f5523eb1ae58ab7afe"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ],
            [
                "versions.yml:md5,57e0ecbc0fd3d1f5cfc9d9ed285826e8",
                "versions.yml:md5,5ad5e88ee4e52ccabeef61dcd368d30b",
                "versions.yml:md5,92f25606f0ee994dac8c00fe49676e5a",
                "versions.yml:md5,9611f602102b5aed0fe2b947f9d2270c",
                "versions.yml:md5,b92d6ad60847882b0089f551c5cfc366",
                "versions.yml:md5,d5e8b4d761dd85a303814239077bd666"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-19T09:47:39.98853048"
    },
    "star - with igenomes - stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.toTranscriptome.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.final.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.Aligned.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.sortedByCoord.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "testXd.out.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test.ReadsPerGene.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.SJ.out.tab:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "test.tab:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            ],
            [
                "versions.yml:md5,5ad5e88ee4e52ccabeef61dcd368d30b",
                "versions.yml:md5,92f25606f0ee994dac8c00fe49676e5a",
                "versions.yml:md5,9611f602102b5aed0fe2b947f9d2270c",
                "versions.yml:md5,b92d6ad60847882b0089f551c5cfc366",
                "versions.yml:md5,d5e8b4d761dd85a303814239077bd666",
                "versions.yml:md5,e3b043ba840d5bddde92f0bf1c91a5fc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-21T14:54:32.959245404"
    }
}